src/main/scala/zio/nio/channels/AsynchronousChannel.scala

```diff
@@ -27,9 +27,13 @@ class AsynchronousByteChannel(private val channel: JAsynchronousByteChannel) {
   final def read(capacity: Int): IO[Exception, Chunk[Byte]] =
     for {
       b <- Buffer.byte(capacity)
-      _ <- readBuffer(b)
+      l <- readBuffer(b)
       a <- b.array
-      r = Chunk.fromArray(a)
+      r <- if (l == -1) {
+            write(Chunk.succeed(0)).const(Chunk.empty)
```

Comment:
Why write a single 0 byte here? I haven't been able to figure it out.

Reply:
damm, I need to push my latest change because I got rid of that. The Idea was that you need to use connection to get exception that is closed already because TCP connection don't know that it's been closed unless you use it. so I send whatever :)

Reply:
@quelgar please look into latest commit.

Reply:
Maybe extract it to some helper function as clean-up.